ios/demo/src/MapViewController.m

```diff
@@ -32,6 +32,22 @@ - (void)mapView:(TGMapViewController *)mapView didLoadSceneAsync:(NSString *)sce
     [mapView setPosition:newYork];
 }
 
+- (void)mapView:(TGMapViewController *)mapView didSelectMarker:(TGMapMarkerId)markerId atScreenPosition:(CGPoint)position
+{
+    if (markerId == 0) {
```

Comment:
We want to always have a callback, it's a nicer behavior. As the implementing developer querying the system without knowing if the query has ever completed is misleading; you actually don't know if it actually performed the query at all. So you never really know if it failed for other reasons than not finding a pick result.